ios/Sources/AutoLayoutView.swift

```diff
@@ -215,4 +224,44 @@ import UIKit
             return (cellFrame.minY >= boundsStart || cellFrame.maxY >= boundsStart) && (cellFrame.minY <= boundsEnd || cellFrame.maxY <= boundsEnd)
         }
     }
+
+    /// Fixes footer position along with rest of the items
+    func fixFooter() {
+		guard !disableAutoLayout, let parentScrollView = getScrollView() else {
+			return
+		}
+
+		let isAutoLayoutEndVisible = horizontal ? frame.maxX <= parentScrollView.frame.width : frame.maxY <= parentScrollView.frame.height
+		guard isAutoLayoutEndVisible, let footer = getFooter() else {
+			return
+		}
+
+		let diff = getFooterDiff()
+		guard diff != 0 else { return }
+
+		if (horizontal) {
+			footer.frame.origin.x += diff
+			frame.size.width += diff
+			superview?.frame.size.width += diff
+		} else {
+			footer.frame.origin.y += diff
+			frame.size.height += diff
+			superview?.frame.size.height += diff
+		}
+	}
+
+    func getFooterDiff() -> CGFloat {
+        if subviews.count == 0 {
+            lastMaxBoundOverall = 0
```

Comment:
clearGaps only runs if you have more than one child. I also wanted to keep footer stuff disconnected and not refactor too much. In case of any issue we just have to stop calling fixFooter()
Lack of unit tests isn't helping :(

Reply:
Agreed, we should add them soon as it makes changes to this file painful and one does not have much confidence they are not breaking anything here.